Services/CO.CDP.Organisation.WebApi/UseCase/UpdateJoinRequestUseCase.cs

```diff
@@ -0,0 +1,89 @@
+using CO.CDP.GovUKNotify;
+using CO.CDP.GovUKNotify.Models;
+using CO.CDP.Organisation.WebApi.Model;
+using CO.CDP.OrganisationInformation.Persistence;
+
+namespace CO.CDP.Organisation.WebApi.UseCase;
+
+public class UpdateJoinRequestUseCase(
+    IOrganisationRepository organisationRepository,
+    IOrganisationJoinRequestRepository requestRepository,
+    IPersonRepository personRepository,
+    IGovUKNotifyApiClient govUKNotifyApiClient,
+    IConfiguration configuration,
+    ILogger<UpdateJoinRequestUseCase> logger)
+    : IUseCase<(Guid organisationId, Guid joinRequestId, UpdateJoinRequest updateJoinRequest), bool>
+{
+    public async Task<bool> Execute((Guid organisationId, Guid joinRequestId, UpdateJoinRequest updateJoinRequest) command)
+    {
+        var organisation = await organisationRepository.Find(command.organisationId)
+            ?? throw new UnknownOrganisationException($"Unknown organisation {command.organisationId}.");
+
+        var joinRequest = await requestRepository.Find(command.joinRequestId, command.organisationId)
+            ?? throw new UnknownOrganisationJoinRequestException($"Unknown organisation join request for org id {command.organisationId} or request id {command.joinRequestId}.");
+
+        var person = await personRepository.Find(command.updateJoinRequest.ReviewedBy)
+            ?? throw new UnknownPersonException($"Unknown person {command.updateJoinRequest.ReviewedBy}.");
+
+        joinRequest.ReviewedById = person.Id;
+        joinRequest.Status = command.updateJoinRequest.status;
+        joinRequest.ReviewedOn = DateTimeOffset.UtcNow;
+
+        if (command.updateJoinRequest.status == OrganisationInformation.OrganisationJoinRequestStatus.Accepted)
+        {
+            organisation.OrganisationPersons.Add(new OrganisationPerson
+            {
+                Person = joinRequest.Person!,
+                Organisation = organisation,
+                Scopes = []
+            });
```

Comment:
You'll also need to add a relationship between the person and the company's tenant, otherwise it won't show up on the user's list. Something like: 

```
person.Tenants.Add(organisation.Tenant);
personRepository.Save(person);
```

Reply:
Actually it would be 
```
joinRequest.Person.Tenants.Add(organisation.Tenant);
personRepository.Save(joinRequest.Person);
```

And you'll also need to add the following line to the OrgRepo to include the tenants
```
.Include(p => p.Tenant)
```

as in

```
    public async Task<Organisation?> Find(Guid organisationId)
    {
        return await context.Organisations
            .Include(p => p.Tenant)
            .Include(p => p.Addresses)
            .ThenInclude(p => p.Address)
            .FirstOrDefaultAsync(t => t.Guid == organisationId);
    }
```

Reply:
Done